pgjdbc/src/main/java/org/postgresql/core/Parser.java

```diff
@@ -1367,4 +1364,43 @@ private int getMatchedPosition(char[] p_sql, int pos) {
       return 0;
     }
   }
+
+  static final class IntCollection {
+    private int[] ints = new int[8];
+    private int index = 0;
+
+    IntCollection() {
+    }
+
+    public void add(int i) {
+      if (index >= ints.length) {
+        // double in size until 1024 in size, then grow by 1.5x
+        final int newLength = ints.length < 1024 ? ints.length << 1 : (ints.length + (ints.length >> 1));
```

Comment:
@bokken , is this "overflow conscious code"?

Reply:
it should be overflow safe enough. the largest this safely supports is an array with ~1.4 billion elements. As of java 9 or 10 (when compact string project implemented), a String can only have 1 billion characters in it.